### PR TITLE
DO-188: partially added custom validation.

### PR DESCRIPTION
**Pull request by ayushmalik2110**

### CODE REVIEW

File: src/utils/common/validators/customValidators.js [hunk 1]
```diff
@@ -43,29 +49,102 @@ const dateValidation = (formData = {}, key) => {
 const ageValidation = (formData = {}, key) => {
   const age = Number(formData[key]);
   if (age === 'NaN') {
-    return { error: 'Please insert a number' };
+    return { error: 'Age must be a number' };
```
Comment (dhruvlahoti) [inline src/utils/common/validators/customValidators.js:52]: Should we keep all of the customValidators file messages in a json or constant file?

Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js:52]: Done

File: src/constants/member/memberConstants.js [hunk 2]
```diff
@@ -21,3 +21,41 @@ export const SOCIAL_LOGO_LINKEDIN = 'https://directory-online-qa.s3.ap-south-1.a
 export const SOCIAL_LOGO_TWITTER = 'https://directory-online-qa.s3.ap-south-1.amazonaws.com/default/images/twitter-5.png';
 
 export const SOCIAL_LOGO_YOUTUBE = 'https://directory-online-qa.s3.ap-south-1.amazonaws.com/default/images/youtube-3.png';
+
+const MAX_CHAR_ERROR_MESSAGE = '${field} cannot have more than ${maxChar} characters.';
```
Comment (mohitsaladiya) [inline src/constants/member/memberConstants.js:25]: Can we move all this messages to json?

File: src/utils/common/validators/customValidators.js [hunk 3]
```diff
@@ -15,11 +16,13 @@ import { REGEX_VALIDATE } from '../../../constants/common/regex';
 const dateValidation = (formData = {}, key) => {
   const enteredDate = formData[key];
   const getDateError = () => ({
-    error: 'Invalid Date',
+    error: VALIDATION_ERRORS.INVALID_DATE_ERROR_MESSAGE,
   });
 
-  if (enteredDate !== undefined && enteredDate !== '' && String(enteredDate)
-    .length === 10) {
+  if (enteredDate !== undefined
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:22]:
```suggestion
  if (enteredDate)
```

File: src/utils/common/validators/customValidators.js [hunk 4]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+/**
+ *
+ * @param {Object} formData
+ * @param {string} key
+ * @return {{error: string}|undefined}
+ */
 const firstNameValidation = (formData = {}, key) => {
   const firstName = formData[key];
+
+  if (firstName === undefined) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:83]: Remove this condition.

File: src/utils/common/validators/customValidators.js [hunk 5]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+    return undefined;
+  }
+  if (!((REGEX_VALIDATE.ALPHABET_WITH_SPACE).test(firstName))) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:86]:
```suggestion
  if (firstName && !((REGEX_VALIDATE.ALPHABET_WITH_SPACE).test(firstName))) {
```

File: src/utils/common/validators/customValidators.js [hunk 6]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+/**
+ *
+ * @param {Object} formData
+ * @param {string} key
+ * @return {{error: string}|undefined}
+ */
+const lastNameValidation = (formData = {}, key) => {
+  const lastName = formData[key];
+
+  if (lastName === undefined) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:121]: Here as well

File: src/utils/common/validators/customValidators.js [hunk 7]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+    return undefined;
+  }
+  if (!((REGEX_VALIDATE.ALPHABET_WITH_SPACE).test(lastName))) {
+    return {
+      error: stringReplace(VALIDATION_ERRORS.PLAIN_STRING_ERROR_MESSAGE, {
+        field: 'Last Name',
+      }),
+    };
+  }
+  if (lastName.length < 3) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:131]: Can we get this 3 with schema?

File: src/utils/common/validators/customValidators.js [hunk 8]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+    return {
+      error: stringReplace(VALIDATION_ERRORS.MIN_CHAR_ERROR_MESSAGE, {
+        field: 'Last Name',
+        minChar: '3',
+      }),
+    };
+  }
+  if (lastName.length > 32) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:139]: Can we get this 32 from schema. Say maxChars

File: src/utils/common/validators/customValidators.js [hunk 9]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+/**
+ *
+ * @param {Object} formData
+ * @return {{error: string}|null}
+ */
 const passwordValidation = (formData = {}) => {
   const { password } = formData;
 
   if (password !== undefined
     && password.length < 6) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:178]: Can we read this 6 from schema. Say minimumPasswordLength

File: src/utils/common/validators/customValidators.js [hunk 10]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+  }
+  if (!((REGEX_VALIDATE.ALPHABET_WITH_SPACE).test(specialization))) {
+    return {
+      error: stringReplace(VALIDATION_ERRORS.PLAIN_STRING_ERROR_MESSAGE, {
+        field: 'Specialization',
+      }),
+    };
+  }
+
+  if (specialization.length > 64) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:241]: Here as well

File: src/utils/common/validators/customValidators.js [hunk 11]
```diff
@@ -43,74 +46,368 @@ const dateValidation = (formData = {}, key) => {
...
+ * @param {string} key
+ * @return {{error: string}|undefined}
+ */
+const streetValidation = (formData, key) => {
+  const street = formData[key];
+
+  if (street === undefined) {
+    return undefined;
+  }
+  if (street.length > 64) {
```
Comment (mohitsaladiya) [inline src/utils/common/validators/customValidators.js:265]: Here too and below

File: config/tenant/ROYAL/routes/meber.json [hunk 12]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (mohitsaladiya) [inline config/tenant/ROYAL/routes/meber.json:1]: Typo in fileName

File: src/constants/member/memberConstants.js (hunk 2 above)
Comment (ayushmalik2110) [reply src/constants/member/memberConstants.js]: done

File: src/utils/common/validators/customValidators.js (hunk 11 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: config/tenant/ROYAL/routes/meber.json (hunk 12 above)
Comment (ayushmalik2110) [reply config/tenant/ROYAL/routes/meber.json:1]: done

File: src/utils/common/validators/customValidators.js (hunk 10 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: src/utils/common/validators/customValidators.js (hunk 9 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: src/utils/common/validators/customValidators.js (hunk 8 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: src/utils/common/validators/customValidators.js (hunk 7 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: src/utils/common/validators/customValidators.js (hunk 6 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: src/utils/common/validators/customValidators.js (hunk 5 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done

File: src/utils/common/validators/customValidators.js (hunk 4 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js]: done.

File: src/utils/common/validators/customValidators.js (hunk 3 above)
Comment (ayushmalik2110) [reply src/utils/common/validators/customValidators.js:22]: done.

**Pull request merged**